drivers/spi/spi_nrfx_spim.c

```diff
@@ -143,12 +149,13 @@ static int configure(const struct device *dev,
 
 	ctx->config = spi_cfg;
 	spi_context_cs_configure(ctx);
+	freq = get_nrf_spim_frequency(get_dev_config(dev)->max_freq,
+				      spi_cfg->frequency);
```

Comment:
Why not
```c
freq = get_nrf_spim_frequency(MIN(spi_cfg->frequency,
                                  get_dev_config(dev)->max_freq));
```
?

Reply:
I preferred to keep all the selection logic inside `get_nrf_spim_frequency`, instead of splitting it across two locations.

Reply:
I does not select anything, just limits input value. I do not see any reason here to change get_nrf_spim_frequency().